shell/platform/darwin/ios/framework/Source/vsync_waiter_ios.mm

```diff
@@ -130,40 +123,37 @@ - (void)onDisplayLink:(CADisplayLink*)link {
 
   recorder->RecordVsync(frame_start_time, frame_target_time);
   if (_allowPauseAfterVsync) {
-    display_link_.get().paused = YES;
+    link.paused = YES;
   }
   callback_(std::move(recorder));
 }
 
 - (void)invalidate {
-  [display_link_.get() invalidate];
+  [_displayLink invalidate];
+  _displayLink = nil;  // Break retain cycle.
 }
 
 - (void)dealloc {
```

Comment:
I would remove this method; it's basically a lie (and was before as well). The display link has a strong reference to this object, so it's impossible* for `dealloc` to be called unless the display link has already been invalidated.

* Unless there's an overrelease, of course, but then we have other problems.